### PR TITLE
Add a local client to mimic cloud storage in a regular file system

### PR DESCRIPTION
**Pull request by TomDLT**
This PR adds a LocalClient, similar to S3Client and GDriveClient.

LocalClient is a client that mimics cloud storage in a regular file system. In particular, it adds files `.meta.json` to store the metadata associated with files.

A LocalClient is useful to backup buckets locally without losing metadata.

### CODE REVIEW

File: cottoncandy/s3client.py [hunk 1]
```diff
@@ -69,6 +69,11 @@ def __init__(self, bucket, access_key, secret_key, s3url, force_bucket_creation=
         self.bucket_name = None
 
         if bucket:
+            if not self.check_bucket_exists(bucket):
+                bucket_new = "glab-" + bucket.replace("_", "-")
+                if self.check_bucket_exists(bucket_new):
+                    bucket = bucket_new
+
```
Comment (anwarnunez) [inline cottoncandy/s3client.py:76]: This is a bit hard-coded. Consider using the config value `mandatory_bucket_prefix` instead
https://github.com/gallantlab/cottoncandy/blob/main/cottoncandy/defaults.cfg#L11

Comment (TomDLT) [reply cottoncandy/s3client.py:76]: Oops, this is a personal hack, not meant to be in this pull-request. I will remove it.

**Comment by TomDLT** [comment]
Tests are passing.

**Comment by anwarnunez** [comment]
If things look go to y'all, then go ahead and merge.

**Pull request merged**